istioctl/cmd/proxyconfig.go

```diff
@@ -116,13 +118,21 @@ var stringToLevel = map[string]Level{
 }
 
 var (
-	loggerLevelString = ""
-	reset             = false
+	loggerLevelString      = ""
+	reset                  = false
+	once                   sync.Once
+	isZtunnelPodKubeClient kube.CLIClient
 )
 
 var isZtunnelPod = func(podName, podNamespace string) bool {
-	client, _ := kubeClient(kubeconfig, configContext)
-	return ambientutil.IsZtunnelPod(client, podName, podNamespace)
+	var err error
+	once.Do(func() {
+		isZtunnelPodKubeClient, err = kubeClient(kubeconfig, configContext)
+		if err != nil {
+			panic(err)
```

Comment:
Let me know if a panic makes sense here or do we want to return? My initial gut is to avoid an extra value being returned `isZtunnelPod` because then we are doing error checks in every call to this and that seems less than ideal.

cc/ @keithmattix @hanxiaop

Reply:
If you're going to panic, try and wrap the `kubeClient` call in some sort of retry (preferably pkg/backoff)